src/base_cmdline.cpp

```diff
@@ -0,0 +1,242 @@
+#include <wx/wxprec.h>
+
+#ifndef WX_PRECOMP
+#  include <wx/wx.h>
+#endif
+
+#include "base.h"
+#include "controller.h"
+#include "misc.h"
+#include "flags.h"
+#include "special_chars.h"
+#include <cassert>
+#include <filesystem>
+#include <sstream>
+
+// contains all command line options
+static const wxCmdLineEntryDesc s_cmd_line_desc[] =
+{
+  { wxCMD_LINE_SWITCH, "h", "help", "Display help for command line interface", wxCMD_LINE_VAL_NONE, wxCMD_LINE_OPTION_HELP},
+  // required
+  { wxCMD_LINE_OPTION, "r", "radius", "Probe radius", wxCMD_LINE_VAL_DOUBLE},
+  { wxCMD_LINE_OPTION, "g", "grid", "Spatial resolution of the underlying grid", wxCMD_LINE_VAL_DOUBLE},
+  { wxCMD_LINE_OPTION, "fs", "file-structure", "Path to the structure file", wxCMD_LINE_VAL_STRING},
+  // optional
+  { wxCMD_LINE_OPTION, "fr", "file-radius", "Path to the radius file", wxCMD_LINE_VAL_STRING},
+  { wxCMD_LINE_OPTION, "do", "dir-output", "Path to the output directory", wxCMD_LINE_VAL_STRING},
+  { wxCMD_LINE_OPTION, "r2", "radius2", "Large probe radius (for two-probe mode)", wxCMD_LINE_VAL_DOUBLE},
+  { wxCMD_LINE_OPTION, "d", "depth", "Octree depth", wxCMD_LINE_VAL_NUMBER},
+  { wxCMD_LINE_SWITCH, "ht", "hetatm", "Include HETATM from pdb file", wxCMD_LINE_VAL_NONE, 0},
+  { wxCMD_LINE_SWITCH, "uc", "unitcell", "Evaluate unit cell", wxCMD_LINE_VAL_NONE, 0},
+  { wxCMD_LINE_SWITCH, "sf", "surface", "Calculate surfaces", wxCMD_LINE_VAL_NONE, 0},
+  { wxCMD_LINE_SWITCH, "xr", "export-report", "Export report (requires:-do)", wxCMD_LINE_VAL_NONE, 0},
+  { wxCMD_LINE_SWITCH, "xt", "export-total", "Export total surface map (requires:-do)", wxCMD_LINE_VAL_NONE, 0},
+  { wxCMD_LINE_SWITCH, "xc", "export-cavities", "Export surface maps for all cavities (requires:-do)", wxCMD_LINE_VAL_NONE, 0},
+  { wxCMD_LINE_OPTION, "o", "output", "Control what parts of the output to display (default:all)", wxCMD_LINE_VAL_STRING},
+  { wxCMD_LINE_SWITCH, "q", "quiet", "Silence progress reporting", wxCMD_LINE_VAL_NONE, 0},
+  { wxCMD_LINE_SWITCH, "ac", "ascii", "Limit output to ascii", wxCMD_LINE_VAL_NONE},
```

Comment:
I would suggest to set ascii mode by default in command line. On windows the chemical formula, angström, superscript symbols are not displayed properly.

Reply:
I was considering this. Will be done.